### ISSUE TITLE
Fatal error: exception (Invalid_argument "equal: abstract value")

### ISSUE THREAD

**Issue by sgrove** [Solo5/solo5#74]
Ultimately probably caused by #73, but posting here in case it's a concrete manifestation of the problem, or if it turns out to be unrelated.

Hardcoding an IP to get far enough in the unikernel boot (sidestepping #72 just for a moment) to see what other issues lie ahead, I run into a `Fatal error: exception (Invalid_argument "equal: abstract value")` caused by the equality comparison on https://github.com/mirleft/ocaml-tls/blob/master/lib/config.ml#L127:

```
SeaBIOS (version 1.8.2-20160510_123855-google)
Total RAM Size = 0x0000000026600000 = 614 MiB
CPU Mhz=2201
CPUs found: 1     Max CPUs supported: 1
found virtio-scsi at 0:3
virtio-scsi vendor='Google' product='PersistentDisk' rev='1' type=0 removable=0
virtio-scsi blksize=512 sectors=2097152 = 1024 MiB
drive 0x000f5500: PCHS=0/0/0 translation=lba LCHS=1024/32/63 s=2097152
Booting from Hard Disk 0...

SYSLINUX 6.03 2014-10-06 Copyright (C) 1994-2014 H. Peter Anvin et al
Loading unikernel.bin... ok
            |      ___|
  __|  _ \  |  _ \ __ \
\__ \ (   | | (   |  ) |
____/\___/ _|\___/____/
multiboot: Using memory: 0x100000 - 0x265fd000
Initializing the KVM Paravirtualized clock.
virtio_config: device_id=8, interrupt_line=b
Found unknown virtio device!
virtio_config: device_id=1, interrupt_line=b
Found virtio network device with MAC: 42 01 0a 8a 00 03 
Solo5: new bindings
STUB: getenv() called
Netif: plugging into tap0 with mac 42:01:0a:8a:00:03
Netif: connect tap0
2016-08-12 15:27.55: INF [ethif] Connected Ethernet interface 42:01:0a:8a:00:03
2016-08-12 15:27.55: INF [arpv4] Connected arpv4 device on 42:01:0a:8a:00:03
2016-08-12 15:27.55: INF [udp] UDP interface connected on 0.0.0.0
STUB: open() called
STUB: getpid() called
virtio-scsi blksize=512 sectors=2097152 = 1024 MiB
STUB: getppid() called
2016-08-12 15:27.55: INF [tcpip-stack-direct] Manager: connect
2016-08-12 15:27.55: INF [tcpip-stack-direct] Manager: configuring
2016-08-12 15:27.55: INF [tcpip-stack-direct] Manager: Interface to 10.0.0.2 nm 255.255.255.0 gw [10.0.0.1]
2016-08-12 15:27.55: INF [tcpip-stack-direct] Manager: configuration done
Fatal error: exception (Invalid_argument "equal: abstract value")
Raised by primitive operation at file "lib/config.ml", line 127, characters 24-34
Called from file "list.ml", line 73, characters 12-15
Called from file "lib/config.ml", line 201, characters 2-57
Called from file "lib/config.ml", line 245, characters 29-51
Called from file "site_dispatch.ml", line 358, characters 15-64
Called from file "site_dispatch.ml", line 365, characters 4-17
Called from file "main.ml", line 254, characters 2-75
Called from file "camlinternalLazy.ml", line 25, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 32, characters 10-11
Called from file "camlinternalLazy.ml", line 25, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 32, characters 10-11
Called from file "src/core/lwt.ml", line 660, characters 16-24
Mirage on Solo5 exiting... Goodbye!
```

**Comment by mato** [comment]
Data point: I can reproduce this with `static_website_tls` current `mirage-dev` builds for `virtio`, but not for `ukvm`.

**Comment by mato** [comment]
Fixed via #76.
